openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -926,19 +942,101 @@ def import_staged_content_from_user_clipboard(library_key: LibraryLocatorV2, use
     if not user_clipboard:
         return None
 
-    olx_str = content_staging_api.get_staged_content_olx(user_clipboard.content.id)
-
-    # TODO: Handle importing over static assets
+    staged_content_id = user_clipboard.content.id
+    olx_str = content_staging_api.get_staged_content_olx(staged_content_id)
+    staged_content_files = content_staging_api.get_staged_content_static_files(staged_content_id)
 
     content_library, usage_key = validate_can_add_block_to_library(
         library_key,
         user_clipboard.content.block_type,
         block_id
     )
 
+    # content_library.learning_package is technically a nullable field because
+    # it was added in a later migration, but we can't actually make a Library
+    # without one at the moment. TODO: fix this at the model level.
+    learning_package: LearningPackage = content_library.learning_package  # type: ignore
+
+    now = datetime.now(tz=timezone.utc)
+
     # Create component for block then populate it with clipboard data
-    _create_component_for_block(content_library, usage_key, user.id)
-    set_library_block_olx(usage_key, olx_str)
+    with transaction.atomic():
+        # First create the Component, but do not initialize it to anything (i.e.
+        # no ComponentVersion).
+        component_type = authoring_api.get_or_create_component_type(
+            "xblock.v1", usage_key.block_type
+        )
+        component = authoring_api.create_component(
+            learning_package.id,
+            component_type=component_type,
+            local_key=usage_key.block_id,
+            created=now,
+            created_by=user.id,
+        )
+
+        # This will create the first component version and set the OLX/title
+        # appropriately. It will not publish. Once we get the newly created
+        # ComponentVersion back from this, we can attach all our files to it.
+        component_version = set_library_block_olx(usage_key, olx_str)
+
+        for staged_content_file_data in staged_content_files:
+            # The ``data`` attribute is going to be None because the clipboard
+            # is optimized to not do redundant file copying when copying/pasting
+            # within the same course (where all the Files and Uploads are
+            # shared). Learning Core backed content Components will always store
+            # a Component-local "copy" of the data, and rely on lower-level
+            # deduplication to happen in the ``contents`` app.
+            filename = staged_content_file_data.filename
+
+            # Grab our byte data for the file...
+            file_data = content_staging_api.get_staged_content_static_file_data(
+                staged_content_id,
+                filename,
+            )
+            if not file_data:
+                log.error(
+                    f"Staged content {staged_content_id} included referenced "
+                    f" file {filename}, but no file data was found."
```

Comment:
```suggestion
                    f"file {filename}, but no file data was found."
```
[nit] previous line already has a space